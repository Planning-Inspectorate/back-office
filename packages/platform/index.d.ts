import './http/hooks';
import './testing/assets/assets';
import './testing/fake';
import './testing/html-parser';
import './utils/date';
import './utils/dotenv';
import './utils/filter';
import './utils/redis';
import './utils/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './http/hooks';
	export * from './testing/assets/assets';
	export * from './testing/fake';
	export * from './testing/html-parser';
	export * from './utils/date';
	export * from './utils/dotenv';
	export * from './utils/filter';
	export * from './utils/redis';
	export * from './utils/url';
	export * from './validators/date';
	export * from './validators/postcode';
}

export * from './types/account';
