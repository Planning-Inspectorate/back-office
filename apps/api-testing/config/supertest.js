// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
