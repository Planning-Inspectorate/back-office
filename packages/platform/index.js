export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
