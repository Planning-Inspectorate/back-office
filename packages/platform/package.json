{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
