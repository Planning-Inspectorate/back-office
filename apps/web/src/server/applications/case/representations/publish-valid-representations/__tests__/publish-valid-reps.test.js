import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../testing/index.js';
import { publishableRepresentationsFixture } from '../../__fixtures__/publishable-representations.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const baseUrl =
	'/applications-service/case/1/relevant-representations/publish-valid-representations';

const mockCaseData = {
	title: 'mock title'
};

describe('publish-valid-representations', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	describe('GET /applications-service/:caseId/relevant-representations/publish-valid-representations', () => {
		describe('and the publishable rep count is > 0', () => {
			const nocks = () => {
				nock('http://test/').get('/applications/1').reply(200, mockCaseData);
				nock('http://test/')
					.get('/applications/1/representations/publishable')
					.reply(200, publishableRepresentationsFixture);
			};
			nocks();

			it('should render the page with the publish button', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('Publish documents');
			});
		});

		describe('and the publishable rep count is 0', () => {
			const nocks = () => {
				nock('http://test/').get('/applications/1').reply(200, mockCaseData);
				nock('http://test/')
					.get('/applications/1/representations/publishable')
					.reply(200, { itemCount: 0 });
			};
			nocks();

			it('should render the page without the publish button', async () => {
				const response = await request.get(baseUrl);
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).not.toContain('Publish documents');
			});
		});
	});

	describe('POST /applications-service/:caseId/relevant-representations/publish-valid-representations', () => {
		const nocks = () => {
			nock('http://test/')
				.get('/applications/1/representations/publishable')
				.reply(200, publishableRepresentationsFixture);
			nock('http://test/')
				.patch('/applications/1/representations/publish')
				.reply(200, { publishedRepIds: [1, 2, 3] });
		};
		beforeEach(async () => {
			nocks();
		});
		describe('and publish action succeeded to publish all valid representations', () => {
			it('should render the page with success banner', async () => {
				const response = await request.post(baseUrl);

				expect(response?.headers?.location).toContain(
					'/applications-service/case/1/relevant-representations?published=3'
				);
			});
		});
	});
});
