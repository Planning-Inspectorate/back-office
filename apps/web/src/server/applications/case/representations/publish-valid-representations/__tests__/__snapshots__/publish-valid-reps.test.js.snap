// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publish-valid-representations GET /applications-service/:caseId/relevant-representations/publish-valid-representations and the publishable rep count is > 0 should render the page with the publish button 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\"><span class=\\"govuk-caption-m\\">Publishing queue</span>
    <h1 class=\\"govuk-heading-l govuk-!-margin-bottom-9\\">Publish all valid representations</h1>
    <p class=\\"govuk-body\\"><strong>Project name: </strong> mock title</p>
    <p class=\\"govuk-body\\">The queue contains <strong>3</strong> representations ready to be published.</p>
    <p     class=\\"govuk-inset-text\\">The representations will not be visible on the external website until
        the key date for their display has been reached.
        <br>
        <br>Key dates are managed by the Case Team.</p>
        <form method=\\"post\\">
            <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Publish documents</button>
        </form>
</main>"
`;

exports[`publish-valid-representations GET /applications-service/:caseId/relevant-representations/publish-valid-representations and the publishable rep count is 0 should render the page without the publish button 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <h1 class=\\"govuk-heading-xl\\">Sorry, there is a problem with the service</h1>
            <p class=\\"govuk-body\\">Try again later.</p>
            <p class=\\"govuk-body\\"><a href=\\"https://has-appeal.herokuapp.com/help/contact\\" class=\\"govuk-link\\">Contact the Planning Inspectorate Customer Support</a> if
                the problem persists.</p>
        </div>
    </div>
</main>"
`;
